### PR TITLE
use cert watcher from operatorkit

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: service/create/service.go [hunk 1]
```diff
@@ -54,37 +55,45 @@ const (
...
+		Logger:      nil,
+		S3Bucket:    "",
+		PubKeyFile:  "",
 	}
 }
 
 // New creates a new configured service.
 func New(config Config) (*Service, error) {
 	// Dependencies.
+	// TODO other deps have to be checked as well
```
Comment (asymmetric) [inline service/create/service.go]: What other deps? Is it worth mentioning them, and/or having a ticket for it?

Comment (xh3b4sd) [inline service/create/service.go]: The config is divided in dependency and settings sections. A dependency is something that implements business logic. A setting is basically data. Logger and k8s client are dependencies required to make the service actually work. Therefore the constructor here should make sure that sufficient configuration is provided. At the moment I can just forget to inject a k8s client and the operator will fail on startup or panic during runtime.

Comment (asymmetric) [inline service/create/service.go]: OK, so it seems we need an issue to tackle this at some point?

Comment (xh3b4sd) [inline service/create/service.go]: See https://github.com/giantswarm/aws-operator/issues/225.

**Comment by xh3b4sd** [comment]
Fishing for LGTM.

**Pull request merged**